src/lib.rs

```diff
@@ -223,9 +223,29 @@ fn components(blurhash: &str) -> Result<(usize, usize), Error> {
     Ok((num_x, num_y))
 }
 
+#[cfg(feature = "image")]
+/// Calculates the blurhash for an [DynamicImage][image::DynamicImage] using the given x and y component counts.
+pub fn encode_image(components_x: u32, components_y: u32, image: &image::DynamicImage) -> Result<String, Error> {
+    use image::{GenericImageView, EncodableLayout};
+    encode(components_x, components_y, image.width(), image.height(), image.to_rgba8().as_bytes())
+}
+
+#[cfg(feature = "image")]
+/// Decodes the given blurhash to an image of the specified size.
+///
+/// The punch parameter can be used to de- or increase the contrast of the
+/// resulting image.
+pub fn decode_image(blurhash: &str, width: u32, height: u32, punch: f32) -> Result<image::DynamicImage, Error> {
+    let bytes = decode(blurhash, width, height, punch)?;
+    // Save to unwrap as `decode` (if successfull) always returns a buffer of size `4 * width * height`, which is exactly
+    // the amount of bytes required to construct the `RgbaImage`.
+    let buffer = image::RgbaImage::from_raw(width, height, bytes).expect("decoded image too small");
+    Ok(image::DynamicImage::ImageRgba8(buffer))
+}
+
 #[cfg(test)]
 mod tests {
-    use super::{decode, encode};
+    use super::*;
     use image::{save_buffer, ColorType::Rgba8};
     use image::{EncodableLayout, GenericImageView};
```

Comment:
These imports should also be feature-gated.

Reply:
Testing already has image enabled by default, so this should not be a problem.